modules/api/pkg/ee/clusterbackup/backupstoragelocation/handler.go

```diff
@@ -0,0 +1,456 @@
+	if err != nil {
+		return nil, err
+	}
+
+	req.GetClusterReq = cr.(cluster.GetClusterReq)
+	return req, nil
+}
+
+func ListBSLEndpoint(ctx context.Context, request interface{}, userInfoGetter provider.UserInfoGetter, projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider, settingsProvider provider.SettingsProvider) (*apiv2.BackupStorageLocationList, error) {
+	if err := validateUser(ctx, userInfoGetter, ""); err != nil {
+		return nil, err
+	}
+	if err := clusterbackup.IsClusterBackupEnabled(ctx, settingsProvider); err != nil {
+		return nil, err
+	}
+
+	req := request.(listBSLReq)
+	if err := validateUser(ctx, userInfoGetter, req.ProjectID); err != nil {
+		return nil, err
+	}
+	client, err := handlercommon.GetClusterClientWithClusterID(ctx, userInfoGetter, projectProvider, privilegedProjectProvider, req.ProjectID, req.ClusterID)
+	if err != nil {
+		return nil, err
+	}
+
+	bslList := &velerov1.BackupStorageLocationList{}
+	if err := client.List(ctx, bslList, ctrlruntimeclient.InNamespace(clusterbackup.UserClusterBackupNamespace)); err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+
+	res := []apiv2.BackupStorageLocationOverview{}
+	for _, bsl := range bslList.Items {
+		res = append(res, apiv2.BackupStorageLocationOverview{
+			Name:         bsl.Name,
+			CreationDate: bsl.ObjectMeta.CreationTimestamp.Time,
+			CBSLName:     bsl.Labels[CBSL],
+			Region:       bsl.Spec.Config[bslRegion],
+			Prefix:       bsl.Spec.ObjectStorage.Prefix,
+			Status:       bsl.Status,
+		})
+	}
+	return &apiv2.BackupStorageLocationList{Items: res}, nil
+}
+
+// deleteBSLReq defines HTTP request for to delete the backup storage location from the cluster specified by the name
+// swagger:parameters deleteBackupStorageLocation
+type deleteBSLReq struct {
+	cluster.GetClusterReq
+
+	// in: path
+	// required: true
+	BSLName string `json:"bsl_name"`
+}
+
+func DecodeDeleteBSLReq(c context.Context, r *http.Request) (interface{}, error) {
+	var req deleteBSLReq
+	cr, err := cluster.DecodeGetClusterReq(c, r)
+	if err != nil {
+		return nil, err
+	}
+	req.GetClusterReq = cr.(cluster.GetClusterReq)
+
+	req.BSLName = mux.Vars(r)["bsl_name"]
+	if req.BSLName == "" {
+		return nil, fmt.Errorf("'bsl_name' parameter is required but was not provided")
+	}
+	return req, nil
+}
+
+func DeleteBSLEndpoint(ctx context.Context, request interface{}, userInfoGetter provider.UserInfoGetter, projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider, settingsProvider provider.SettingsProvider) (interface{}, error) {
+	if err := validateUser(ctx, userInfoGetter, ""); err != nil {
+		return nil, err
+	}
+	if err := clusterbackup.IsClusterBackupEnabled(ctx, settingsProvider); err != nil {
+		return nil, err
+	}
+
+	req := request.(deleteBSLReq)
+	if err := validateUser(ctx, userInfoGetter, req.ProjectID); err != nil {
+		return nil, err
+	}
```

Comment:
No need to call the function twice if you already know from the beginning what needs to be validated for the user.